core/src/main/java/org/apache/calcite/jdbc/CalciteSchema.java

```diff
@@ -71,6 +74,9 @@ public abstract class CalciteSchema {
   protected final NameSet functionNames;
   protected final NameMap<FunctionEntry> nullaryFunctionMap;
   protected final NameMap<CalciteSchema> subSchemaMap;
+
+  // tableName -> (user, access)
+  protected final NameMap<Map<Principal, SqlAccessType>> privilegeMap;
```

Comment:
Whether to use a separate class to maintain the relationship between `Principal`, `SqlAccessType`, and `resource_type(TABLE/SCHEMA/FUNCTION...)`. 
Using this way resource types can be more general and facilitate subsequent expansion.